aten/src/ATen/native/Pow.cpp

```diff
@@ -118,7 +126,11 @@ Tensor& float_power_(Tensor& base, const Tensor& exp) {
 }
 
 Tensor& float_power_(Tensor& base, Scalar exp) {
-  return base.float_power_(c10::scalar_to_tensor(exp, base.device()));
+  auto dtype = (at::isComplexType(base.scalar_type()) || exp.isComplex()) ? at::kComplexDouble : at::kDouble;
+  TORCH_CHECK(base.scalar_type() == dtype,
+              "self tensor type ", base.scalar_type(), "is not the desired type ", dtype);
```

Comment:
We're inconsistent in our checks (note: we should pick a style an be consistent) but typically want to mention the operation and use the name in the docs, not "self" which is typically an internal name. Also, "tensor type" is ambiguous and this should refer to dtype.

Maybe an error message like:

"the base given to float_power_ has dtype DTYPE, but the operation's result requires dtype DTYPE"

?

Reply:
Note also there's not a space between the first `base.scalar_type()` and `"is` in the string. A similar printing error can be seen in the error messages below.

Reply:
is this even correct behavior?  Don't we generally check that the computation type is castable to the result type, not that they are the same?

Reply:
Good point. I extended the Developer FAQ entry to address this explicitly (see [here](https://github.com/pytorch/pytorch/wiki/Developer-FAQ#how-does-out-work-in-pytorch)). 

So yes, a corrected version of the check would validate that the operation's common dtype is "safe castable" to base's dtype.

Reply:
In this case, delegating to the out= variant of pow here would probably save adding special logic for this case. That logic needs to be added to the out= variant (per other comments), anyway.

Reply:
Ok, since this is a different issue, I filed https://github.com/pytorch/pytorch/issues/50213.